### ISSUE TITLE
Understanding the detokenization approach for distillation

### ISSUE THREAD

**Issue by rm-asif-amin** [openvla/openvla#290]
Hi !
I'm attempting to distill the model, for which I'd need the soft labels. So I attempted to extract soft labels in a Libero setting. The raw soft labels(logits) shape for a single inference is (1, 34, 32064).    32064 is the vocab size of the llama tokenizer.

- Can anyone please help me understand why the second dimension is 34?
-  My current understanding is, for each time-step, I should extract 256 tokens for each action dimension. Should taking logits[:,-7:, -256:]  work? 
- The paper mentions an action de-tokonizer module. How can I find the module logic?

Thanks in Advance

Here's the relevant part from the paper for reference-
> we discretize each dimension of the robot actions separately into one of 256 bins. For each action dimension, we set the bin width to uniformly divide the interval between the 1st and 99th quantile of the actions in the training data. Using quantiles instead of the min-max bounds Brohan et al. [7] used allows us to ignore outlier actions in the data that could otherwise drastically expand the discretization interval and reduce the effective granularity of our action discretization.
> Using this discretization, we obtain N discrete integers ∈ [0 . . . 255] for an N -dimensional robot ac- tion. Unfortunately, the tokenizer used by OpenVLA’s language backbone, the Llama tokenizer [10], only reserves 100 “special tokens” for tokens newly introduced during fine-tuning, which is too few for the 256 tokens of our action discretization. Instead, we again opt for simplicity and follow Brohan et al. [7]’s approach by simply overwriting the 256 least used tokens in the Llama tokenizer’s vocabulary (which corresponds to the last 256 tokens) with our action tokens. Once the actions are processed into a sequence of tokens, OpenVLA is trained with a standard next-token prediction objective, evaluating the cross-entropy loss on the predicted action tokens only.
